src/redux/actions/filterData.js

```diff
@@ -1,12 +1,19 @@
 import { updateFilteredData, updateSearchKey } from './actionCreators';
 
-const filterData = (searchKey, data) => {
+const filterData = (searchKey, data, onlyFavourites, favouriteItems) => {
```

Comment:
Use object as a single argument with given fields as key - remember to update dispatch method of this action

Reply:
U don't need to send through arguments data from reducers.

return (dispatch) <--- contains second argument ---> getState() ! ;)